From f3057aad862482625384f1402e494e12b637f2ed Mon Sep 17 00:00:00 2001
From: Myy <myy@miouyouyou.fr>
Date: Mon, 16 Jan 2017 12:44:56 +0000
Subject: [PATCH] Readapt: ARM: dts: rockchip: miqi: add turbo-mode operating
 points

Readaptation of Willy Tarreau patch.

Here's the original commit message:

By switching to opp-v2 we can declare "turbo-mode" operating points
which are only enabled when /sys/devices/system/cpu/cpufreq/boost is
set. It is convenient because it allows to boot, set a safe powersave
governor, enable boost, limit scaling_max_freq to a safe value, then
change the governor to performance or ondemand, and the frequency can
then be manually adjusted by only touching scaling_max_freq.

New values are 1896, 1920, 1992, 2016, 2040 MHz, 2064, 2088, 2112,
2136, 2160, 2184, 2208. MiQi boards work fine up to 2112 with a very
good power supply (5.2V/3A real) and a strong heatsink. Higher
frequencies may randomly work. At least 1992 is rock solid for hours
using "openssl speed -multi 4". The other ones have only been tested
for a few minutes. Frequencies of 1896 and 1920 MHz use 1.425V.
1992 MHz uses 1.45V. 2016, 2040 and 2064 use 1.475V. 2088 and above
use 1.500V. 2160, 2184 and 2208 cause the lowest frequency to be
picked. It's obvious that it's a sign issue somewhere in the kernel
but this one was not found yet.

Signed-off-by: Myy <myy@miouyouyou.fr>

diff -Naur a/arch/arm/boot/dts/rk3288-firefly.dtsi b/arch/arm/boot/dts/rk3288-firefly.dtsi
--- a/arch/arm/boot/dts/rk3288-firefly.dtsi	2017-06-25 01:25:48.272282361 +0300
+++ b/arch/arm/boot/dts/rk3288-firefly.dtsi	2017-06-25 01:32:54.180298688 +0300
@@ -296,6 +296,7 @@
 
 &cpu0 {
 	cpu0-supply = <&vdd_cpu>;
+	operating-points-v2 = <&cpu0_opp_table>;
 };
 
 &edp {
diff -Naur a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
--- a/arch/arm/boot/dts/rk3288.dtsi	2017-06-25 01:28:23.821288324 +0300
+++ b/arch/arm/boot/dts/rk3288.dtsi	2017-06-25 01:36:01.179305856 +0300
@@ -175,15 +175,82 @@
 			opp-microvolt = <1200000>;
 			clock-latency-ns = <40000>;
 		};
-		opp-1512000000 {
+		opp@1512000000 {
 			opp-hz = /bits/ 64 <1512000000>;
-			opp-microvolt = <1300000>;
-			clock-latency-ns = <40000>;
+			opp-microvolt = <1250000>;
 		};
-		opp-1608000000 {
+		opp@1608000000 {
 			opp-hz = /bits/ 64 <1608000000>;
+			opp-microvolt = <1300000>;
+		};
+		opp@1704000000 {
+			opp-hz = /bits/ 64 <1704000000>;
 			opp-microvolt = <1350000>;
-			clock-latency-ns = <40000>;
+		};
+		opp@1800000000 {
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1400000>;
+		};
+		/* boot-only frequencies below */
+		opp@1896000000 {
+			opp-hz = /bits/ 64 <1896000000>;
+			opp-microvolt = <1425000>;
+			turbo-mode;
+		};
+		opp@1920000000 {
+			opp-hz = /bits/ 64 <1920000000>;
+			opp-microvolt = <1425000>;
+			turbo-mode;
+		};
+		opp@1992000000 {
+			opp-hz = /bits/ 64 <1992000000>;
+			opp-microvolt = <1450000>;
+			turbo-mode;
+		};
+		opp@2016000000 {
+			opp-hz = /bits/ 64 <2016000000>;
+			opp-microvolt = <1475000>;
+			turbo-mode;
+		};
+		opp@2040000000 {
+			opp-hz = /bits/ 64 <2040000000>;
+			opp-microvolt = <1475000>;
+			turbo-mode;
+		};
+		opp@2064000000 {
+			opp-hz = /bits/ 64 <2064000000>;
+			opp-microvolt = <1475000>;
+			turbo-mode;
+		};
+		opp@2088000000 {
+			opp-hz = /bits/ 64 <2088000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
+		opp@2112000000 {
+			opp-hz = /bits/ 64 <2112000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
+		opp@2136000000 {
+			opp-hz = /bits/ 64 <2136000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
+		opp@2160000000 {
+			opp-hz = /bits/ 64 <2160000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
+		opp@2184000000 {
+			opp-hz = /bits/ 64 <2184000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
+		opp@2208000000 {
+			opp-hz = /bits/ 64 <2208000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
 		};
 	};
 
